docs/mkdocs/docs/integration/migration_guide.md

```diff
@@ -0,0 +1,248 @@
+
+        ```cpp
+        nlohmann::json j;
+        std::stringstream ss("[1,2,3]");
+        ss >> j;
+        ```
+
+- Passing iterator pairs or pointer/length pairs to parsing functions ([`parse`](../api/basic_json/parse.md),
+  [`accept`](../api/basic_json/accept.md), [`sax_parse`](../api/basic_json/sax_parse.md),
+  [`from_cbor`](../api/basic_json/from_cbor.md), [`from_msgpack`](../api/basic_json/from_msgpack.md),
+  [`from_ubjson`](../api/basic_json/from_ubjson.md), and [`from_bson`](../api/basic_json/from_bson.md) via initializer
+  lists is deprecated since 3.8.0. Instead, pass two iterators; for instance, call `from_cbor(ptr, ptr+len)` instead of
+  `from_cbor({ptr, len})`.
+
+    === "Deprecated"
+  
+          ```cpp
+          const char* s = "[1,2,3]";
+          bool ok = nlohmann::json::accept({s, s + std::strlen(s)});
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          const char* s = "[1,2,3]";
+          bool ok = nlohmann::json::accept(s, s + std::strlen(s));
+          ```
+
+#### JSON Pointers
+
+- Comparing JSON Pointers with strings via [`operator==`](../api/json_pointer/operator_eq.md) and
+  [`operator!=`](../api/json_pointer/operator_ne.md) is deprecated since 3.11.2. To compare a
+  [`json_pointer`](../api/json_pointer/index.md) `p` with a string `s`, convert `s` to a `json_pointer` first and use
+  [`json_pointer::operator==`](../api/json_pointer/operator_eq.md) or
+  [`json_pointer::operator!=`](../api/json_pointer/operator_ne.md).
+    
+    === "Deprecated"
+  
+        ```cpp
+        nlohmann::json::json_pointer lhs("/foo/bar/1");
+        assert(lhs == "/foo/bar/1");
+        ```
+  
+    === "Future-proof"
+
+        ```cpp
+        nlohmann::json::json_pointer lhs("/foo/bar/1");
+        assert(lhs == nlohmann::json::json_pointer("/foo/bar/1"));
+        ```
+
+- The implicit conversion from JSON Pointers to string
+  ([`json_pointer::operator string_t`](../api/json_pointer/operator_string_t.md)) is deprecated since 3.11.0. Use
+  [`json_pointer::to_string`](../api/json_pointer/to_string.md) instead.
+  
+    === "Deprecated"
+  
+          ```cpp
+          nlohmann::json::json_pointer ptr("/foo/bar/1");
+          std::string s = ptr;
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          nlohmann::json::json_pointer ptr("/foo/bar/1");
+          std::string s = ptr.to_string();
+          ```
+
+- Passing a `basic_json` specialization as template parameter `RefStringType` to
+  [`json_pointer`](../api/json_pointer/index.md) is deprecated since 3.11.0. The string type can now be directly
+  provided.
+  
+    === "Deprecated"
+  
+          ```cpp
+          using my_json = nlohmann::basic_json<std::map, std::vector, my_string_type>;
+          nlohmann::json_pointer<my_json> ptr("/foo/bar/1");
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          nlohmann::json_pointer<my_string_type> ptr("/foo/bar/1");
+          ```
```

Comment:
You could also mention, that `nlohmann::my_json::json_pointer` is an alias for `nlohmann::json_pointer<my_string_type>` and is always an alias to the `json_pointer` with the appropriate string type for all specializations of `basic_json`.

Reply:
Done.